src/ast/dibuilderbpf.cpp

```diff
@@ -59,14 +86,182 @@ DIType *DIBuilderBPF::getInt64Ty()
   return types_.int64;
 }
 
+DIType *DIBuilderBPF::getInt128Ty()
+{
+  if (!types_.int128)
+    types_.int128 = createBasicType("int128", 128, dwarf::DW_ATE_signed);
+
+  return types_.int128;
+}
+
+DIType *DIBuilderBPF::getIntTy()
+{
+  if (!types_.int_)
+    types_.int_ = createBasicType("int", 32, dwarf::DW_ATE_signed);
+
+  return types_.int_;
+}
+
 DIType *DIBuilderBPF::getInt8PtrTy()
 {
   if (!types_.int8_ptr)
-    types_.int8_ptr = createPointerType(
-        createBasicType("int8", 8, dwarf::DW_ATE_signed), 64);
+    types_.int8_ptr = createPointerType(getInt8Ty(), 64);
 
   return types_.int8_ptr;
 }
 
+// Create anonymous struct with anonymous fields. It's possible that there will
+// be multiple tuples of the same (duplicated) type but BTF deduplication should
```

Comment:
If it's not deduped, will it go over the instruction limit with duplicated types or does that not count toward that?

Reply:
I don't think that this has anything to do with instruction limit. This is just a DWARF entry which will be transformed into a BTF entry.